File: PA18.java
class Solution {
    public int longestPalindrome(String s) {
        if(s == null || s.length()==0){
            return 0;
        }
    int count=0;
     HashSet<Character> hs = new HashSet<>();
        for(int i=0; i<s.length();i++){
            if(hs.contains(s.charAt(i))){
                count=count+2;
                hs.remove(s.charAt(i));
            }
            else{
                hs.add(s.charAt(i));
            }
        }
        if(hs.size()>0){
            return count+1;
        }
        return count;
        
    }
}

File: PA_18.py
#LC 409 - Longest Palindrome
#Time Complexity - O(n)
#Space Complexity - O(n)
class Solution(object):
    def longestPalindrome(self, st):
        """
        :type s: str
        :rtype: int
        """
        if len(st) == 1:
            return 1
        s = set()
        count = 0
        for i in st:
            if i in s:
                s.remove(i)
                count = count + 2
            else:
                s.add(i)
        if len(s) > 0:
            return count + 1
        return count